src/Microsoft.SqlTools.Migration/MigrationService.cs

```diff
@@ -580,6 +583,62 @@ public void InitializeService(IProtocolEndpoint serviceHost)
             }
         }
 
+        internal async Task HandleGetArmTemplateRequest(
+    string targetType,
+    RequestContext<string> requestContext)
+        {
+            ProvisioningScriptServiceProvider provider = new ProvisioningScriptServiceProvider();
+
+            List<SkuRecommendationResult> recommendations = ExtractSkuRecommendationReportAction.ExtractSkuRecommendationsFromReport(GetReportFilePath(SqlAssessmentConfiguration.ReportsAndLogsRootFolderPath, targetType));
+
+            SqlArmTemplate template = provider.GenerateProvisioningScript(recommendations);
+
+            string jsonOutput = JsonConvert.SerializeObject(
+                template,
+                Formatting.Indented,
+                new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore }
+            );
+
+            await requestContext.SendResult(jsonOutput);
+
+        }
+
+        /// <summary>
+        /// Scans the specified output folder for a SkuRecommendationReport*.json file, used when no report file path is specified 
+        /// </summary>
+        /// <param name="outputFolder"></param>
+        /// <returns>The file path to the SKU recommendation report, if it exists</returns>
+        public static string GetReportFilePath(string outputFolder, string targetType)
+        {
+            string skuRecommendationJsonRegex = "SkuRecommendationReport-";
+            switch (targetType)
+            {
+                case "AzureSqlDatabase":
+                    skuRecommendationJsonRegex += "AzureSqlDatabase-Baseline*.json";
```

Comment:
Here we need to pick latest recommendation, the way this line is ready it may choose any file. And same for other targets.

Reply:
fixed . now from ADS in the request we are supplying the path of the latest recommendation.